### DISCUSSION TITLE
How to enable Adasum in single node multi gpu environment build with NCCL?

### DISCUSSION THREAD

**Discussion by FanFanFanYang** [horovod/horovod#2775] [Q&A]
**Environment:**
1. Framework: TensorFlow
2. Framework version: 1.15.2
3. Horovod version: 0.20.3
4. MPI version:4.0.2
5. CUDA version:10.0.0
6. NCCL version:2.4
7. Python version:3.7
8. OS and version: Amazon Linux AMI
9. GCC version:7.3.0

**Your question:**
As mentioned in the doc, if build with NCCL, then Adasum is not supported on single node. https://github.com/horovod/horovod/blob/master/docs/adasum_user_guide.rst#key-takeaways
Is it possible to enable Adasum in single node? Can I build Horovod without NCCL? Thank you.

**Comment by tgaddair** [comment]
Hey @FanFanFanYang, if NCCL is not enabled, then Adasum will work correctly on a single node (meaning, you do not need to scale your learning rate with the number of worker processes).

**Comment by FanFanFanYang** [comment]
@tgaddair Thanks. I'm wondering how to disable NCCL? Compile without NCCL flag? Is there anything bad if not use NCCL?

**Comment by tgaddair** [comment]
Hey @FanFanFanYang, training speed will be decreased without NCCL in most cases.  You can disable NCCL by not setting `HOROVOD_GPU_OPERATIONS` during install.
